crates/weaver_live_check/src/attribute_live_check.rs

```diff
@@ -0,0 +1,727 @@
+                    });
+                }
+            }
+        }
+
+        // run advisors on the attribute
+        for advisor in self.advisors.iter_mut() {
+            if let Ok(advices) = advisor.advise(sample_attribute, semconv_attribute.as_ref()) {
+                for advice in advices {
+                    attribute_result.add_advice(advice);
+                }
+            }
+        }
+
+        attribute_result
+    }
+
+    /// Run advisors on every attribute in the list
+    #[must_use]
+    pub fn check_attributes(&mut self, sample_attributes: Vec<SampleAttribute>) -> LiveCheckReport {
+        let mut live_check_report = LiveCheckReport {
+            attributes: Vec::new(),
+            statistics: LiveCheckStatistics::new(),
+        };
+
+        for sample_attribute in sample_attributes.iter() {
+            let attribute_result = self.create_live_check_attribute(sample_attribute);
+
+            // Update statistics
+            live_check_report.statistics.update(&attribute_result);
+            live_check_report.attributes.push(attribute_result);
+        }
+        live_check_report
+    }
+}
+
+/// Represents a live check attribute parsed from any source
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct LiveCheckAttribute {
+    /// The sample attribute
+    pub sample_attribute: SampleAttribute,
+    /// Advice on the attribute
+    pub all_advice: Vec<Advice>,
+    /// The highest advisory level
+    pub highest_advisory: Option<Advisory>,
+}
+
+impl LiveCheckAttribute {
+    /// Create a new LiveCheckAttribute
+    #[must_use]
+    pub fn new(sample_attribute: SampleAttribute) -> Self {
+        LiveCheckAttribute {
+            sample_attribute,
+            all_advice: Vec::new(),
+            highest_advisory: None,
+        }
+    }
+
+    /// Add an advice to the attribute and update the highest advisory level
+    pub fn add_advice(&mut self, advice: Advice) {
+        let advisory = advice.advisory.clone();
+        if let Some(previous_highest) = &self.highest_advisory {
+            if previous_highest < &advisory {
+                self.highest_advisory = Some(advisory);
+            }
+        } else {
+            self.highest_advisory = Some(advisory);
+        }
+        self.all_advice.push(advice);
+    }
+}
+
+/// A live check report for a set of attributes
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct LiveCheckReport {
+    /// The live check attributes
+    pub attributes: Vec<LiveCheckAttribute>,
+    /// The statistics for the report
+    pub statistics: LiveCheckStatistics,
+}
+
+impl LiveCheckReport {
+    /// Return true if there are any violations in the report
+    #[must_use]
+    pub fn has_violations(&self) -> bool {
+        self.statistics
+            .highest_advisory_counts
+            .contains_key(&Advisory::Violation)
+    }
+}
+
+/// The statistics for a live check report
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct LiveCheckStatistics {
```

Comment:
As mentioned earlier, I think we should add information to obtain or calculate coverage details for attributes (and, in the future, for signals) relative to the reference registry.